pinecone/grpc/vector_factory_grpc.py

```diff
@@ -0,0 +1,122 @@
+import numbers
+
+from collections.abc import Iterable, Mapping
+from typing import Union, Tuple, Dict
+
+from google.protobuf.struct_pb2 import Struct
+
+from .utils import dict_to_proto_struct
+from ..utils import fix_tuple_length, convert_to_list
+from ..utils.constants import REQUIRED_VECTOR_FIELDS, OPTIONAL_VECTOR_FIELDS
+from ..data import (
+    VectorDictionaryMissingKeysError, 
+    VectorDictionaryExcessKeysError, 
+    VectorTupleLengthError, 
+    SparseValuesTypeError, 
+    SparseValuesMissingKeysError, 
+    SparseValuesDictionaryExpectedError, 
+    MetadataDictionaryExpectedError 
+)
+
+from pinecone.core.grpc.protos.vector_service_pb2 import (
+    Vector as GRPCVector,
+    QueryVector as GRPCQueryVector,
+    SparseValues as GRPCSparseValues,
+)
+from pinecone import (
+    Vector as NonGRPCVector, 
+    SparseValues as NonGRPCSparseValues
+)
+
+class VectorFactoryGRPC:
```

Comment:
Thanks for all the work getting the grpc side of things a bit more aligned with the non-grpc paths. 🙏